packages/runtime/container-runtime/README.md

```diff
@@ -51,6 +51,54 @@ issue](https://github.com/microsoft/FluidFramework/wiki/Submitting-Bugs-and-Feat
 
+## Data Virtualization For DataStores
+
+This is the ability of container to delay load datastores at any point in time by asking for the datastore snapshot at
+anytime, not caring about from where it is coming and then using that to load that datastore and return it to the app.
+
+### How Current System Works
+
+Fluid (through its APIs) does expose some levels of data virtualization. But under covers, there is almost none. The
+whole content of the file is downloaded in one go. Due to limitation of data virtualization, FF holds ALL blobs in
+snapshot as those might be required in the future. Any delayed loading (through FF APIs) results in loading state of
+DDSs at a sequence number of snapshot we booted from, and application of all the ops up till current sequence number.
+While application may choose not to load some data stores immediately on boot (and realize some saving in time and
+memory by not allocating appropriate app state for such datastores), FF still pays the costs for such content.
+It also continues to pay the cost for all such content indefinitely, even if those datastores were loaded (this could
+theoretically be optimized, but now there is no good way for driver to learn what blobs are safe to discard due to
+limitations in interface and lack of appropriate semantics at driver API layer).
+
+### Improvement with Data Virtualization
+
+With this, we will provide a capability to:
+
+- Exclude some sub-trees from snapshot payload, thus allowing faster transfer times / boot times and smaller initial
+memory footprint.
+- Ability to safely (in current system, i.e., without introducing any new breaking behaviors) to delay-load data
+stores later.
+
+Container Runtime Apis like IContainerRuntimeBase.createDataStore and IContainerRuntimeBase.createDetachedDataStore
+provides an argument `dataStoreGroupIdForSnapshotFetch` which allows apps to put a datastore in group at time of
+creation currently. This represents the group of the datastore within a container or its snapshot. When not specified
+the datastore will belong to a `default` group. When a container is loaded initially, only datastores which belongs to
+`default` group are fetched from service and can be loaded on demand when requested by user. This decreases the amount
+of data which needs to be fetched during load and hence provides faster boot times for the container. Snapshot for all
+datastores within a non-default datastore groupId will be fetched from service when any of the datastores within a
+group is requested by application, then snapshot for that particular group will be fetched using a network call at that
+time. So, 1 network will be required to fetch snapshot for a group when a datastore from a group is requested by an
+application. This allows applications to put a datastore in a group, when it thinks that, that particular datastore
+will not be requested on boot and will be loaded later on like on some user action like a button click. This will
```

Comment:
I'd refrain from using words "allows" and "when it thinks".
I'd use something like "when datastore is assigned to a group, content of such data store will not be loaded with initial load of container. It will be loaded only when any datastore with such groupID is realized.

Reply:
same below - let's remove all mentions to "thinks" :)

Reply:
done